test/general/TestFill.py

```diff
@@ -273,3 +278,167 @@ def test_multiplayer_rules_fill_restrictive(self):
+        self.assertEqual(locations[1].item, basic_items[1])
+        self.assertEqual(locations[2].item, basic_items[0])
+        self.assertEqual(locations[3].item, prog_items[0])
+
+    def test_too_many_excluded_distribute(self):
+        multi_world = generate_multi_world()
+        player1 = generate_player_data(
+            multi_world, 1, 4, prog_item_count=2, basic_item_count=2)
+        locations = player1.locations
+
+        locations[0].progress_type = LocationProgressType.EXCLUDED
+        locations[1].progress_type = LocationProgressType.EXCLUDED
+        locations[2].progress_type = LocationProgressType.EXCLUDED
+
+        self.assertRaises(FillError, distribute_items_restrictive, multi_world)
+
+    def test_non_excluded_item_must_distribute(self):
+        multi_world = generate_multi_world()
+        player1 = generate_player_data(
+            multi_world, 1, 4, prog_item_count=2, basic_item_count=2)
+        locations = player1.locations
+        basic_items = player1.basic_items
+
+        locations[1].progress_type = LocationProgressType.EXCLUDED
+        locations[2].progress_type = LocationProgressType.EXCLUDED
+        basic_items[0].never_exclude = True
+        basic_items[1].never_exclude = True
+
+        self.assertRaises(FillError, distribute_items_restrictive, multi_world)
+
+    def test_priority_distribute(self):
+        multi_world = generate_multi_world()
+        player1 = generate_player_data(
+            multi_world, 1, 4, prog_item_count=2, basic_item_count=2)
+        locations = player1.locations
+        prog_items = player1.prog_items
+        basic_items = player1.basic_items
+
+        locations[0].progress_type = LocationProgressType.PRIORITY
+        locations[3].progress_type = LocationProgressType.PRIORITY
+
+        distribute_items_restrictive(multi_world)
+
+        self.assertEqual(locations[0].item, prog_items[1])
+        self.assertEqual(locations[1].item, basic_items[1])
+        self.assertEqual(locations[2].item, basic_items[0])
+        self.assertEqual(locations[3].item, prog_items[0])
+
+    def test_excess_priority_distribute(self):
+        multi_world = generate_multi_world()
+        player1 = generate_player_data(
+            multi_world, 1, 4, prog_item_count=2, basic_item_count=2)
+        locations = player1.locations
+        prog_items = player1.prog_items
+        basic_items = player1.basic_items
+
+        locations[0].progress_type = LocationProgressType.PRIORITY
+        locations[1].progress_type = LocationProgressType.PRIORITY
+        locations[2].progress_type = LocationProgressType.PRIORITY
+
+        distribute_items_restrictive(multi_world)
+
+        self.assertEqual(locations[0].item, basic_items[0])
+        self.assertEqual(locations[1].item, prog_items[0])
+        self.assertEqual(locations[2].item, prog_items[1])
+        self.assertEqual(locations[3].item, basic_items[1])
+
+    def test_multiple_world_distribute(self):
```

Comment:
These two tests are helpful for illustrative purposes, but are also kind of brittle. If we change anything about our randomization, the asserts need to be regenerated. I mostly added them to show what the distribution of items look like. It would be pretty easy to make test_multiple_world_priority_distribute a more robust test, but it would lose its illustrative nature.